File: samples/client/3_0_3_unit_test/java/src/main/java/org/openapijsonschematools/client/schemas/validation/ConstValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.exceptions.ValidationException;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.math.BigDecimal;
import java.util.List;
import java.util.Objects;

public class ConstValidator extends BigDecimalValidator implements KeywordValidator {
    public final @Nullable Object constValue;

    public ConstValidator(@Nullable Object constValue) {
        this.constValue = constValue;
    }

    @Override
    public @Nullable PathToSchemasMap validate(
        JsonSchema schema,
        @Nullable Object arg,
        ValidationMetadata validationMetadata,
        List<PathToSchemasMap> containsPathToSchemas
    ) {
        if (arg instanceof Number) {
            BigDecimal castArg = getBigDecimal((Number) arg);
            if (Objects.equals(castArg, constValue)) {
                return null;
            }
            if (constValue instanceof BigDecimal && ((BigDecimal) constValue).compareTo(castArg) == 0) {
                return null;
            }
        } else {
            if (Objects.equals(arg, constValue)) {
                return null;
            }
        }
        throw new ValidationException("Invalid value "+arg+" was not equal to const "+constValue);
    }
}

File: samples/client/3_0_3_unit_test/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/ConstValidator.kt
package org.openapijsonschematools.client.schemas.validation

import org.openapijsonschematools.client.exceptions.ValidationException

import java.math.BigDecimal

class ConstValidator : BigDecimalValidator(), KeywordValidator {
    @Throws(ValidationException::class)
    override fun validate(
        data: ValidationData
    ): PathToSchemasMap? {
        if (data.schema.constValue == null) {
            return null
        }
        val constValue: Any? = data.schema.constValue.getData()
        if (data.arg is Number) {
            val castArg = getBigDecimal(data.arg)
            if (castArg == constValue) {
                return null
            }
            if (constValue is BigDecimal && constValue.compareTo(castArg) == 0) {
                return null
            }
        } else {
            if (data.arg == constValue) {
                return null
            }
        }
        throw ValidationException("Invalid value " + data.arg + " was not equal to const " + constValue)
    }
}